README.md

```diff
@@ -26,31 +64,53 @@ Scripts for installation via popular package managers such as Homebrew, Scoop or
 
 K0sctl is still in an early stage of development. Missing major features include at least:
 
-* Cluster upgrades are not yet possible
 * Windows targets are not yet supported
 * The released binaries have not been signed
-* Cluster uninstall and host clean up after failure is not there yet
 * Cluster backup and restore are not available yet
+* Nodes can't be removed
 
 ## Usage
 
-The main function of k0sctl is the `k0sctl apply` subcommand. Provided a configuration file, k0sctl will connect to the listed hosts and install k0s on them.
+### `k0sctl apply`
+
+The main function of k0sctl is the `k0sctl apply` subcommand. Provided a configuration file describing the desired cluster state, k0sctl will connect to the listed hosts, determines the current state of the hosts and configures them as needed to form a k0s cluster.
 
 The default location for the configuration file is `k0sctl.yaml` in the current working directory. To load a configuration from a different location, use:
 
-```
+```sh
 $ k0sctl apply --config path/to/k0sctl.yaml
 ```
 
-## Configuration file syntax
+If the configuration cluster version `spec.k0s.version` is greater than the version detected on the cluster, a cluster upgrade will be performed. If the configuration lists hosts that are not part of the cluster, they will be configured to run k0s and will be joined to the cluster.
 
-To generate a simple skeleton configuration file, you can use the `k0sctl init` subcommand:
+### `k0sctl init`
 
-```
-$ k0sctl init > k0sctl.yaml
+Generate a configuration template. Use `--k0s` to include an example `spec.k0s.config` k0s configuration block.
+
+### `k0sctl reset`
+
+Uninstall k0s from the hosts listed in the configuration.
+
+### `k0sctl kubeconfig`
+
+Connects to the cluster and outputs a kubeconfig file that can be used with `kubectl` or `kubeadm` to manage the kubernetes cluster.
+
+Example:
+
+```sh
+$ k0sctl kubeconfig --config path/to/k0sctl.yaml > k0s.config
+$ kubectl get node --kubeconfig k0s.config
+NAME      STATUS     ROLES    AGE   VERSION
+worker0   NotReady   <none>   10s   v1.20.2-k0s1
 ```
 
-### Example configuration
+## Configuration file
```

Comment:
This section does not look very nice.

Reply:
I don't have any ideas right now how to improve on that, should probably use the k0sproject site for documentation in the future.